lib/combine_pdf/page_methods.rb

```diff
@@ -63,15 +63,16 @@ def inject_page(obj, top = true)
       # injecting each of the values in the injected Page
       res = resources
       obj.resources.each do |key, new_val|
-        unless PDF::PRIVATE_HASH_KEYS.include? key # keep CombinePDF structual data intact.
-          if res[key].nil?
-            res[key] = new_val
-          elsif res[key].is_a?(Hash) && new_val.is_a?(Hash)
-            new_val = new_val[:referenced_object] || new_val
-            new_val.update (res[key][:referenced_object] || res[key]) # make sure the old values are respected
-            (res[key][:referenced_object] || res[key]).update new_val # transfer old and new values to the injected page
-          end #Do nothing if array - ot is the PROC array, which is an issue
-        end
+        # keep CombinePDF structural data intact.
+        next if PDF::PRIVATE_HASH_KEYS.include?(key)
```

Comment:
You're saying Rubocop messed up my logic... again... 😖 I'll go look at the pre-beautification logic. Thanks.